float32[360] angles
float32[360] distances
uint8  iter
